src/actions/channels.js

```diff
@@ -381,30 +381,25 @@ export function getChannelMembers(channelId, page = 0, perPage = General.CHANNEL
 
 export function leaveChannel(channelId) {
     return async (dispatch, getState) => {
-        dispatch({type: ChannelTypes.LEAVE_CHANNEL_REQUEST}, getState);
-
-        const {currentUserId} = getState().entities.users;
-
-        try {
-            await Client4.removeFromChannel(currentUserId, channelId);
-        } catch (error) {
-            forceLogoutIfNecessary(error, dispatch);
-            dispatch(batchActions([
-                {type: ChannelTypes.LEAVE_CHANNEL_FAILURE, error},
-                getLogErrorAction(error)
-            ]), getState);
-            return;
-        }
+        const state = getState();
+        const {currentUserId} = state.entities.users;
+        const {channels} = state.entities.channels;
+        const channel = channels[channelId];
 
-        dispatch(batchActions([
-            {
-                type: ChannelTypes.LEAVE_CHANNEL,
-                data: {id: channelId, user_id: currentUserId}
-            },
-            {
-                type: ChannelTypes.LEAVE_CHANNEL_SUCCESS
+        dispatch({
+            type: ChannelTypes.LEAVE_CHANNEL,
+            data: {id: channelId, user_id: currentUserId},
+            meta: {
+                offline: {
+                    effect: () => Client4.removeFromChannel(currentUserId, channelId),
+                    commit: {type: ChannelTypes.LEAVE_CHANNEL},
+                    rollback: {
+                        type: ChannelTypes.RECEIVED_CHANNEL,
+                        data: channel
```

Comment:
Join channel eventually dispatches `RECEIVED_CHANNEL` so this is basically the same thing, but what is not happening here is `RECEIVED_MY_CHANNEL_MEMBER` so this won't work, I'll need to make the change